### DISCUSSION TITLE
Missing Value (NaNs) on target

### DISCUSSION THREAD

**Discussion by davitens** [Nixtla/neuralforecast#1521] [Q&A]
When there are are NaNs on target and interpolate is not possible, which algorithm can be used for forecasting?

**Comment by omni-front** [comment]
Could you provide more details about the version of the `neuralforecast` library you're using? Different versions might have different capabilities or recommended practices for dealing with NaNs. Additionally, are you looking for a specific type of model or any general recommendations within the `neuralforecast` framework? This would help in suggesting an appropriate approach or model to handle NaNs when interpolation isn't feasible.

**Comment by amarakramali** [comment]
There is not a particular forecasting algorithm that makes raw `NaN` targets valid. The current NeuralForecast quickstart requires observed `y` values to be numeric and non-missing, while the core supports an `available_mask` for timestamps whose target is genuinely unavailable.

If a missing value means “unknown” rather than a real zero, first create the complete regular time grid for each `unique_id`, then mask the unknown targets and replace them with a harmless placeholder:

```python
observed = df['y'].notna()
df['available_mask'] = observed.astype('float32')
df.loc[~observed, 'y'] = 0.0

nf.fit(df=df)
```

Rows with `available_mask == 0` are excluded from the training loss. Filling those masked `y` values is still important because the target tensor itself must not contain `NaN`s. This lets you use models such as NHITS or NBEATS without inventing interpolated labels; model choice should then depend on the horizon, seasonality, amount of observed history, and exogenous inputs rather than on the missing-value mechanism.

Two distinctions matter:

- If the value is a genuine zero (for example, confirmed zero demand), use `y = 0` with `available_mask = 1` because it is an observed target.
- If `y` is merely unknown in the future, that is the normal forecasting case; provide future exogenous variables through `futr_df`, but do not add unknown future targets to the training data.

With long or frequent gaps, also shorten `input_size` or add reliable exogenous features so that sampled windows still contain enough observed history. The mask prevents invalid loss terms, but it cannot create information that was never observed.

Relevant implementation: [`NeuralForecast._check_nan`](https://github.com/Nixtla/neuralforecast/blob/main/neuralforecast/core.py) checks only rows marked available, and the dataset/model window interface carries the availability mask through training.
